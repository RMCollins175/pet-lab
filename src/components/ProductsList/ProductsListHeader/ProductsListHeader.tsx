export const ProductsListHeader = () => {
  return (
    <li className="grid justify-center grid-cols-[0.2fr_1.5fr_0.5fr_0.5fr] p-4 border-b border-gray-300 cursor-pointer">
      <div></div>
      <span
        className="flex items-center justify-center"
        aria-label="Title"
        role="columnheader">
        <strong>Title</strong>
      </span>

      <span
        className="flex items-center justify-center"
        aria-label="Price"
        role="columnheader">
        <strong>Price</strong>
      </span>

      <span
        className="flex items-center justify-center"
        aria-label="Discount"
        role="columnheader">
        <strong>Discount</strong>
      </span>
    </li>
  );
};
