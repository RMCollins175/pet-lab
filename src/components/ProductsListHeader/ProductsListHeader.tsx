interface ProductsListHeaderProps {
  numberOfProdcuts: number;
}

export const ProductsListHeader = ({
  numberOfProdcuts,
}: ProductsListHeaderProps) => {
  return (
    <li className="grid justify-center grid-cols-[0.2fr_1.5fr_0.5fr_0.5fr] p-4 border-b border-gray-300 cursor-pointer">
      <th
        className="flex items-center justify-center text-nowrap"
        aria-label="Title"
        role="columnheader">
        {numberOfProdcuts} Products
      </th>
      <th
        className="flex items-center justify-center"
        aria-label="Title"
        role="columnheader">
        <strong>Title</strong>
      </th>

      <th
        className="flex items-center justify-center"
        aria-label="Price"
        role="columnheader">
        <strong>Price</strong>
      </th>

      <th
        className="flex items-center justify-center"
        aria-label="Discount"
        role="columnheader">
        <strong>Discount</strong>
      </th>
    </li>
  );
};
