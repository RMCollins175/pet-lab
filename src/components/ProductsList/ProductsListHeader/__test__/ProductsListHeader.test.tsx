import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { ProductsListHeader } from "../ProductsListHeader";

describe("ProductsListHeader Component", () => {
  test("renders headers correctly", () => {
    render(<ProductsListHeader />);

    // Using getByRole or getByText to find elements
    const headers = screen.getAllByRole("columnheader"); // if you use heading roles in your headers
    expect(headers[0]).toHaveTextContent("Title");
    expect(headers[1]).toHaveTextContent("Price");
    expect(headers[2]).toHaveTextContent("Discount");

    // Alternatively, use getByText for more specific queries
    expect(screen.getByText("Title")).toBeInTheDocument();
    expect(screen.getByText("Price")).toBeInTheDocument();
    expect(screen.getByText("Discount")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<ProductsListHeader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
