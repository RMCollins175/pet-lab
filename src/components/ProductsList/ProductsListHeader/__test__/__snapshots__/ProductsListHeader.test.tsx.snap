// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsListHeader Component matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="grid justify-center grid-cols-[0.2fr_1.5fr_0.5fr_0.5fr] p-4 border-b border-gray-300 cursor-pointer"
  >
    <div />
    <span
      aria-label="Title"
      class="flex items-center justify-center"
      role="columnheader"
    >
      <strong>
        Title
      </strong>
    </span>
    <span
      aria-label="Price"
      class="flex items-center justify-center"
      role="columnheader"
    >
      <strong>
        Price
      </strong>
    </span>
    <span
      aria-label="Discount"
      class="flex items-center justify-center"
      role="columnheader"
    >
      <strong>
        Discount
      </strong>
    </span>
  </li>
</DocumentFragment>
`;
